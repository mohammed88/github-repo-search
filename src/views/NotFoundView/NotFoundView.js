import React, {Fragment} from 'react';
import styled from 'styled-components';

import Header from '../../components/Header/';
import Footer from '../../components/Footer/';
import {H1, Text18} from '../../ui/styles/typography';
import Link from '../../ui/Link';
import Spacing from '../../ui/styles/spacing';
import GitHubText from '../../ui/icons/github-text';
import {darkOrange} from '../../ui/styles/colors';

const Body = styled.div`
	box-sizing: border-box;
	height: 800px;
	padding: 200px;
	line-height: 1.8;
`;

const Message = styled.div`
	text-align: center;
	width: 400px;
	margin: auto;
	${Text18};
`;

const HomeLink = styled(Link)`
	${Text18};
`;

const NotFoundHeader = styled(H1)`
	margin: ${Spacing.large} 0;
`

const GitHubLogo = styled(GitHubText)`
	fill: ${darkOrange};
`;

function NotFoundView() {
	return (
		<Fragment>
			<Header />
			<Body data-component-type="PageNotFoundMessage">
				<Message>
					<GitHubLogo width={200} height={75}/>
					<NotFoundHeader>404 - Not Found</NotFoundHeader>
					Sorry, the page you are looking for does not exist.
					You can always go back to the <HomeLink href="/">homepage.</HomeLink>
				</Message>
			</Body>
			<Footer />
		</Fragment>
	);
}

export default NotFoundView;