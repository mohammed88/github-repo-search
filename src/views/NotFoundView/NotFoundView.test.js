import React from 'react';
import {shallow} from 'enzyme';

import NotFoundView from './NotFoundView';

describe('NotFoundView', () => {
	it('should render', () => {
		const wrapper = shallow(<NotFoundView />);

		expect(wrapper).toMatchSnapshot() ;
	});
});