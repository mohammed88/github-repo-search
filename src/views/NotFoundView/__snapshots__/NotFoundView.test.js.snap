// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundView should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotFoundView />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Header />,
        <ForwardRef
          data-component-type="PageNotFoundMessage"
        >
          <ForwardRef>
            <ForwardRef
              height={75}
              width={200}
            />
            <ForwardRef>
              404 - Not Found
            </ForwardRef>
            Sorry, the page you are looking for does not exist. You can always go back to the 
            <ForwardRef
              href="/"
            >
              homepage.
            </ForwardRef>
          </ForwardRef>
        </ForwardRef>,
        <Footer />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef>
            <ForwardRef
              height={75}
              width={200}
            />
            <ForwardRef>
              404 - Not Found
            </ForwardRef>
            Sorry, the page you are looking for does not exist. You can always go back to the 
            <ForwardRef
              href="/"
            >
              homepage.
            </ForwardRef>
          </ForwardRef>,
          "data-component-type": "PageNotFoundMessage",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef
                height={75}
                width={200}
              />,
              <ForwardRef>
                404 - Not Found
              </ForwardRef>,
              "Sorry, the page you are looking for does not exist. You can always go back to the ",
              <ForwardRef
                href="/"
              >
                homepage.
              </ForwardRef>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "height": 75,
                "width": 200,
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-gZMcBi",
                  "isStatic": true,
                  "rules": Array [
                    "
	fill: ",
                    "#D37324",
                    ";
",
                  ],
                },
                "displayName": "Styled(GitHubText)",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-gZMcBi",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "404 - Not Found",
              },
              "ref": null,
              "rendered": "404 - Not Found",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-iwsKbI",
                  "isStatic": true,
                  "rules": Array [
                    "
	font-family: ",
                    "'Roboto', sans-serif",
                    ";
	font-size: 32px;
	font-weight: bold;
	margin: 0;
",
                    "
	margin: ",
                    "24px",
                    " 0;
",
                  ],
                },
                "displayName": "Styled(styled.h1)",
                "foldedComponentIds": Array [
                  "sc-bxivhb",
                ],
                "render": [Function],
                "styledComponentId": "sc-iwsKbI",
                "target": "h1",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            "Sorry, the page you are looking for does not exist. You can always go back to the ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "homepage.",
                "href": "/",
              },
              "ref": null,
              "rendered": "homepage.",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-dnqmqq",
                  "isStatic": true,
                  "rules": Array [
                    "
	",
                    "
  color: ",
                    "#EA7F28",
                    ";
  font-size: 1rem;
  text-decoration: none;

  :hover {
  	text-decoration: underline;
    cursor: pointer;
  }
",
                    "
",
                    "
	",
                    "
	font-family: ",
                    "'Roboto', sans-serif",
                    ";
	font-size: 18px;
",
                    ";
",
                  ],
                },
                "displayName": "Styled(styled.a)",
                "foldedComponentIds": Array [
                  "sc-bZQynM",
                ],
                "render": [Function],
                "styledComponentId": "sc-dnqmqq",
                "target": "a",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htoDjs",
              "isStatic": true,
              "rules": Array [
                "
	text-align: center;
	width: 400px;
	margin: auto;
	",
                "
	font-family: ",
                "'Roboto', sans-serif",
                ";
	font-size: 18px;
",
                ";
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-htoDjs",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-gzVnrw",
            "isStatic": true,
            "rules": Array [
              "
	box-sizing: border-box;
	height: 800px;
	padding: 200px;
	line-height: 1.8;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-gzVnrw",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Header />,
          <ForwardRef
            data-component-type="PageNotFoundMessage"
          >
            <ForwardRef>
              <ForwardRef
                height={75}
                width={200}
              />
              <ForwardRef>
                404 - Not Found
              </ForwardRef>
              Sorry, the page you are looking for does not exist. You can always go back to the 
              <ForwardRef
                href="/"
              >
                homepage.
              </ForwardRef>
            </ForwardRef>
          </ForwardRef>,
          <Footer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ForwardRef>
              <ForwardRef
                height={75}
                width={200}
              />
              <ForwardRef>
                404 - Not Found
              </ForwardRef>
              Sorry, the page you are looking for does not exist. You can always go back to the 
              <ForwardRef
                href="/"
              >
                homepage.
              </ForwardRef>
            </ForwardRef>,
            "data-component-type": "PageNotFoundMessage",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <ForwardRef
                  height={75}
                  width={200}
                />,
                <ForwardRef>
                  404 - Not Found
                </ForwardRef>,
                "Sorry, the page you are looking for does not exist. You can always go back to the ",
                <ForwardRef
                  href="/"
                >
                  homepage.
                </ForwardRef>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "height": 75,
                  "width": 200,
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-gZMcBi",
                    "isStatic": true,
                    "rules": Array [
                      "
	fill: ",
                      "#D37324",
                      ";
",
                    ],
                  },
                  "displayName": "Styled(GitHubText)",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-gZMcBi",
                  "target": [Function],
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "404 - Not Found",
                },
                "ref": null,
                "rendered": "404 - Not Found",
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-iwsKbI",
                    "isStatic": true,
                    "rules": Array [
                      "
	font-family: ",
                      "'Roboto', sans-serif",
                      ";
	font-size: 32px;
	font-weight: bold;
	margin: 0;
",
                      "
	margin: ",
                      "24px",
                      " 0;
",
                    ],
                  },
                  "displayName": "Styled(styled.h1)",
                  "foldedComponentIds": Array [
                    "sc-bxivhb",
                  ],
                  "render": [Function],
                  "styledComponentId": "sc-iwsKbI",
                  "target": "h1",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
              "Sorry, the page you are looking for does not exist. You can always go back to the ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "homepage.",
                  "href": "/",
                },
                "ref": null,
                "rendered": "homepage.",
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-dnqmqq",
                    "isStatic": true,
                    "rules": Array [
                      "
	",
                      "
  color: ",
                      "#EA7F28",
                      ";
  font-size: 1rem;
  text-decoration: none;

  :hover {
  	text-decoration: underline;
    cursor: pointer;
  }
",
                      "
",
                      "
	",
                      "
	font-family: ",
                      "'Roboto', sans-serif",
                      ";
	font-size: 18px;
",
                      ";
",
                    ],
                  },
                  "displayName": "Styled(styled.a)",
                  "foldedComponentIds": Array [
                    "sc-bZQynM",
                  ],
                  "render": [Function],
                  "styledComponentId": "sc-dnqmqq",
                  "target": "a",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-htoDjs",
                "isStatic": true,
                "rules": Array [
                  "
	text-align: center;
	width: 400px;
	margin: auto;
	",
                  "
	font-family: ",
                  "'Roboto', sans-serif",
                  ";
	font-size: 18px;
",
                  ";
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-htoDjs",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-gzVnrw",
              "isStatic": true,
              "rules": Array [
                "
	box-sizing: border-box;
	height: 800px;
	padding: 200px;
	line-height: 1.8;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-gzVnrw",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
